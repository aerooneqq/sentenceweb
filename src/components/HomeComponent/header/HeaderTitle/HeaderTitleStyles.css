#headerTitle {
    display: flex;
    flex-direction: row;
    align-items: center;

    opacity: 0;

    animation-name: titleFadeIn;
    animation-duration: 1s;
    animation-delay: 0.4s;
    animation-fill-mode: forwards;
}

    @keyframes titleFadeIn {
        from {
            opacity: 0;
        }

        to {
            opacity: 1;
        }
    }

#headerTitleText { 
    color: #ffffff;
    font-family: "Courier New";
    letter-spacing: 3px;
    font-size: 30px;
    cursor: default;
    visibility: visible;
}

