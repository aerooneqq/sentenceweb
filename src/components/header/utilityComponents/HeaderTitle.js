import React from "react"
import "../styles/HeaderStyles.css"

export default class HeaderTitle extends React.Component {
    constructor() {
        super()

        this.state = {
          titleID: "headerTitleID"
        }
    }

    componentDidMount() {
        var title = document.getElementById(this.state.titleID)
        title.classList.toggle("fadeIn")
    }

    render() {
        return (
            <h2 id={this.state.titleID} className="headerTitle">Sentence</h2>
        )
    }
}
