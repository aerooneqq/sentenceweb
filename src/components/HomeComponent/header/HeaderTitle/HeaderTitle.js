import React from "react"

//Styles
import "./HeaderTitleStyles.css"

export default class HeaderTitle extends React.Component {
    render() {
        return (
            <h2 id = "headerTitle">
                <div id = "headerTitleText">
                    Sentence
                </div>
            </h2>
        )
    }
}
