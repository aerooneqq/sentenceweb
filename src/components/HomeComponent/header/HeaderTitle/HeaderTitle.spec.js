import React from 'react'
import { shallow } from 'enzyme'
import {shallowToJson} from "enzyme-to-json"
import HeaderTitle from './HeaderTitle';

describe("Header title", () => { 
    it("should render correctly", () => { 
        const headerTitle = shallow(<HeaderTitle />);

        expect(shallowToJson(headerTitle)).toMatchSnapshot();
    });

    it("should have 'Sentence' heading", () => { 
        const headerTitle = shallow(<HeaderTitle />);

        expect(headerTitle.find("div").at(0).text()).toBe("Sentence");
    });
});