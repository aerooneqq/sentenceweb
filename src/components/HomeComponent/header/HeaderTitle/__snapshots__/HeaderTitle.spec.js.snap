// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header title should render correctly 1`] = `
<h2
  id="headerTitle"
>
  <div
    id="headerTitleText"
  >
    Sentence
  </div>
</h2>
`;
